bip-xxxx.mediawiki

```diff
@@ -0,0 +1,283 @@
+<pre>
+  BIP: ?
+  Layer: Applications
+  Title: A Simple Payjoin Proposal
+  Author: Nicolas Dorier <nicolas.dorier@gmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-X
+  Status: Draft
+  Type: Standards Track
+  Created: 2019-05-01
+  License: BSD-2-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a protocol for two parties
+to negotiate a coinjoin transaction during a payment between them.
+
+===Copyright===
+
+This BIP is licensed under the 2-clause BSD license.
+
+===Motivation===
+
+When two parties (later referred to as sender and receiver) want to transact,
+most of the time, the sender creates a transaction spending his own Unspent Transaction Outputs (UTXO), signs
+it and broadcasts it on the network.
+
+This simple model gave birth to several heuristics impacting the privacy of the parties and of the network as a whole.
+
+* Common input ownership heuristic: In most transactions, all the inputs belong to the same party.
+* Change identification from scriptPubKey type: If all inputs are spending UTXOs of a certain scriptPubKey type, then the change output is likely to have the same scriptPubKey type, too.
+* Change identification from round amount: If an output in the transaction has a round amount, it is likely an output belonging to the receiver.
+
+We will designate these three heuristics as <code>common-input</code>, <code>change-scriptpubkey</code>, <code>change-round-amount</code>.
+
+The problems we aim to solve are:
+* For the receiver, there is a missed opportunity to consolidate their own UTXO or making payment in the sender's transaction.
+* For the sender, there is privacy leaks about their wallet happening when somebody applies the heuristics detailed above to their transaction.
+
+Our proposal gives an opportunity for the receiver to consolidate their UTXOs and batch their own payment without creating a new transaction. (Saving fee in the process)
+For the sender, it allows them to invalidate the three heuristics above. With the receiver's involvement, the heuristics can even be poisoned. (ie, using the heuristics to intentionally mislead blockchain analysis)
+
+Note that the existence of this proposal is also improving the privacy of parties who are not using it by making the three heuristics unreliable to the network as a whole.
+
+==Specification==
+
+===Protocol===
+In a payjoin payment, the following steps happen:
+
+* The receiver of the payment, present a [[bip-021.mediawiki|BIP 21 URI]] to the sender with a parameter <code>pj</code> describing a http link to the payjoin endpoint.
```

Comment:
Wouldn't it be a good idea to make use of https mandatory?

Reply:
no, because then sender implementation would check for it, making it impossible to work on Tor for people using their libraries.

Our sender at btcpayserver support Tor to allow end user wallet to receive payjoin transactions without configuring NAT  firewall and all...

Reply:
Ahh, yes, http on .onion hidden service is ok. On clearnet there is possible man-in-the-middle attack vector if using plain http.

Reply:
Right, but https or onion are acceptable choices while clearnet http is not. Payjoin would be a degraded protocol by being clearnet considering that it gives away input utxo ownership to the eavesdropper.

Reply:
Well, duh it also ... gives away that it's a Payjoin at all :)

Reply:
ACK, https on clearnet should be mandatory, (unless using regtest/testnet?).

Reply:
I will add a note about it.